app/src/main/java/com/android/streetworkapp/ui/event/EventOverview.kt

```diff
@@ -230,6 +236,50 @@ fun DashBoardBar() {
       }
 }
 
+@Composable
+fun JoinEventButton(
+    event: Event,
+    eventViewModel: EventViewModel,
+    user: User,
+    navigationActions: NavigationActions
+) {
+  val context = LocalContext.current
+
+  Button(
+      onClick = {
+        Toast.makeText(context, "You have joined this event", Toast.LENGTH_LONG).show()
+        eventViewModel.addParticipantToEvent(event.eid, user.uid)
+        navigationActions.goBack()
+      },
+      modifier = Modifier.testTag("joinEventButton"),
+      enabled = event.participants < event.maxParticipants,
+      colors = ColorPalette.BUTTON_COLOR) {
+        Text("Join this event")
+      }
+}
+
+@Composable
+fun LeaveEventButton(
+    event: Event,
+    eventViewModel: EventViewModel,
+    user: User,
+    navigationActions: NavigationActions
+) {
+  val context = LocalContext.current
+
+  Button(
+      onClick = {
+        Toast.makeText(context, "You have left this event", Toast.LENGTH_LONG).show()
+        eventViewModel.removeParticipantFromEvent(event.eid, user.uid)
+        navigationActions.goBack()
+      },
+      enabled = event.owner != user.uid,
+      modifier = Modifier.testTag("leaveEventButton"),
+      colors = ColorPalette.BUTTON_COLOR.copy(containerColor = Color.Red)) {
```

Comment:
Suggestion: I don't think the color of this button is appropriate. It seems to me that bright red is used to display errors, so if I were you I'd use the classic button color. And if you want to draw the user's attention, you can create an alert when he clicks with `Dialog`. What do you think ?

Reply:
The goal of making the button to leave an event ("cancel" action so to speak) red was essentially to differentiate it from the button to join an event ("enter" action)
I think making the buttons two different colors improves readability and helps alert users that the button's action has changed even before he clicks on it. The "cancel" button of the rating popup on the park overview screen uses red for example so i don't think we need to strictly follow a rule of red being used only for errors (as long as we don't go overboard)
That is why i think making both buttons a different color is necessary, let me know what you think!